### PR TITLE
refactor: rewrite Callout for web, add portalling

### PR DESCRIPTION
**Pull request by tigranpetrossian**
Adds a callout for mobile, updates the web version.

### Web
image

* Added ability to display global callouts in the extension using portalling. `<Callout portal />` will render the callout into a corresponding `div`.
* Added `maxContentWidth` prop, to be able to center and limit the content width of global callouts.
* Refactored the component, added a relevant `role`.

### Mobile

https://github.com/user-attachments/assets/e19c40ab-ed51-4a90-ad5d-6f3a64d1bd65

### CODE REVIEW

File: packages/ui/src/components/callout/callout.web.tsx [hunk 1]
```diff
@@ -31,38 +31,71 @@ const calloutRecipe = cva({
...
+    <Box className={calloutRecipe({ variant })} role="status" {...rest}>
+      <Flex
+        justifyContent="space-between"
+        maxWidth={maxContentWidth}
+        mx="auto"
         p="space.05"
-        reverse
-        spacing="space.03"
+        gap="space.03"
+        color="ink.text-primary"
```
Comment (pete-watters) [inline packages/ui/src/components/callout/callout.web.tsx:82]: Do we always need this color or should it be over-writable?

File: packages/ui/src/components/callout/callout.web.tsx [hunk 2]
```diff
@@ -31,38 +31,71 @@ const calloutRecipe = cva({
...
+  success: <CheckmarkCircleIcon />,
+  warning: <ErrorTriangleIcon />,
+  error: <ErrorTriangleIcon />,
+  default: null,
+};
 
-interface CalloutProps extends BoxProps {
-  icon?: ReactNode;
+interface CalloutOwnProps {
+  /** Title of the callout.  */
```
Comment (pete-watters) [inline packages/ui/src/components/callout/callout.web.tsx:45]: I don't think all of these comments are necessary

**Review by pete-watters — APPROVED** [review]
Nice work Tigran 👍  I added some questions / suggestions. 

Generally we like using the `Flag` pattern to keep things more consistent so I'm not sure if removing that is an improvement here

File: packages/ui/panda.config.ts [hunk 3]
```diff
@@ -17,6 +17,16 @@ export default defineConfig({
 
   strictTokens: false,
 
+  staticCss: {
+    css: [
+      {
+        properties: {
```
[the remark this answers is not in the document]
Comment (tigranpetrossian) [reply packages/ui/panda.config.ts:23]: Prop renaming [isn't supported by panda](https://panda-css.com/docs/concepts/style-props#property-renaming) unfortunately, so they added a way to specify custom properties. It's a bit of a crutch, I don't really like the disconnect of having to setup something like this, but decided to chose DX (avoiding confusion over what `maxWidth` is and not overriding the actual `maxWidth`) over generating few bytes of CSS.

File: packages/ui/src/components/callout/callout.web.tsx (hunk 1 above)
Comment (tigranpetrossian) [reply packages/ui/src/components/callout/callout.web.tsx:82]: This was carried over from before the refactor (just different location). We do need it to display the color correctly. Happens to over-writable too but incidentally.

**Comment by tigranpetrossian** [comment]
> Nice work Tigran 👍 I added some questions / suggestions.
> 
> Generally we like using the `Flag` pattern to keep things more consistent so I'm not sure if removing that is an improvement here

@pete-watters The usage of `Flag` here was problematic, in that it used `reverse` for purely visual reasons, which [can be problematic](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction#accessibility)

File: packages/ui/panda.config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tigranpetrossian) [reply packages/ui/panda.config.ts:23]: Oh and this totally should be in the preset, thanks! I actually have no idea how this worked without being in the preset 😳

File: packages/ui/src/components/callout/callout.web.tsx (hunk 2 above)
Comment (pete-watters) [reply packages/ui/src/components/callout/callout.web.tsx:45]: I think that here the only useful comment is:
```
Limit the content width to the specified value. Suitable for full-width callouts displayed at the top of the page.
```

The rest are self explanatory. If you want to add docs I would add them into the `.story` file instead, maybe in the docs description?

```
const meta: Meta<typeof Component> = {
  component: Component,
  tags: ['autodocs'],
  title: 'Callout',
  parameters: {
    docs: {
      description: { 
          component: '...',
      },
```

**Pull request merged**